gobblin-runtime/src/main/java/org/apache/gobblin/runtime/kafka/HighLevelConsumer.java

```diff
@@ -197,6 +201,22 @@ protected void shutdownMetrics() throws IOException {
    */
   protected void createMetrics() {
     this.messagesRead = this.metricContext.counter(RuntimeMetrics.GOBBLIN_KAFKA_HIGH_LEVEL_CONSUMER_MESSAGES_READ);
+    this.queueSizeGauges = new ContextAwareGauge[numThreads];
+    this.queueSizeGaugeValues = new AtomicIntegerArray(numThreads);
+    for (int i=0; i < numThreads; i++) {
+      int finalI = i;
+      this.queueSizeGauges[i] = this.metricContext.newContextAwareGauge(
+          RuntimeMetrics.GOBBLIN_KAFKA_HIGH_LEVEL_CONSUMER_QUEUE_SIZE + "-" + i,
+          () -> queueSizeGaugeValues.get(finalI));
```

Comment:
w/o a comment explaining the reason, I'm left to wonder: why couldn't this be written:
```
() -> queues[finalI].size()
```
?  that would dispense w/ the need for `queueSizeGaugeValues` and with it the potential perf hit of updating that array upon every enqueing.

and on that topic... why is `updateQueueSizes` only recalculated at enqueue time, rather than at dequeue time as well?

Reply:
updating to use `queues[finalI].size()` to avoid having to call a function to explicitly update this